File: Mongo-Nodejs-M101JS/Week 4/HW4.5/answer.js
var mongo = require('mongodb')

mongo.connect('mongodb://localhost:27017/m101',function(err,db){
	var profile = db.collection('profile');

	var _query = {'ns':/school2/i};
	var _sort = {'millis':-1};
	var cursor = profile.find(_query);
	cursor = cursor.sort(_sort);
	cursor.nextObject(function(err,obj){
		console.log("The latency of the longest running operation to the collection " + obj.millis + " ms ");
		return db.close();
	});

});

File: Mongo-Developer-M101P-Pyramid/Week 4/HW4.4/answer.py
import sys, pymongo
from pymongo import MongoClient
from bson.son import SON
import re
connection = MongoClient('localhost', 27017)

db = connection.m101
profile = db.profile
_reg = re.compile('school2')
items = profile.find({'ns':_reg}).sort([('millis',-1)])
for i in items:
	print 'The latency of the longest running operation to the collection %s ms' % i['millis']
	break
